tensorflow/core/util/gpu_solvers.h

```diff
@@ -267,7 +273,49 @@ class GpuSolver {
   Status Trsm(rocblas_side side, rocblas_fill uplo, rocblas_operation trans,
               rocblas_diagonal diag, int m, int n, const Scalar* alpha,
               const Scalar* A, int lda, Scalar* B, int ldb);
-#else  // GOOGLE_CUDA
+
+  // QR factorization.
+  // Computes QR factorization A = Q * R.
+  // Returns Status::OK() if the kernel was launched successfully.
+  // See: http://docs.nvidia.com/cuda/cusolver/#cuds-lt-t-gt-geqrf
+  template <typename Scalar>
+  Status Geqrf(int m, int n, Scalar* dev_A, int lda, Scalar* dev_tau,
+               int* dev_lapack_info);
+
+
+  // This function performs the matrix-matrix addition/transposition
+  //   C = alpha * op(A) + beta * op(B).
+  // Returns Status::OK() if the kernel was launched successfully.  See:
+  // http://docs.nvidia.com/cuda/cublas/index.html#cublas-lt-t-gt-geam
```

Comment:
Was this code moved from elsewhere? If it's ROCm-based, why does it reference nvidia docs?

Reply:
@cheshire The function declarations in gpu_solvers is for both sides. Originally, this file was cuda_solvers.h but it was changed a few months ago to gpu_solvers.h to better represent the dual functionality. These were probably artifacts of those changes. I'll remove the references to cuBLAS docs.

Reply:
@cheshire @stevenireeves  cleaned up the comments.